### PR TITLE
Remove-DbaDatabase

### PR DESCRIPTION
**Pull request by ctrlbold**

### CODE REVIEW

File: functions/Remove-DbaDatabase.ps1 [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[object[]]$SqlInstance,
+		[parameter(Mandatory = $false)]
+		[object]$SqlCredential
+	)
+	
+	DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SqlCredential } }
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: With the advent of TEPP, not necessary anymore

File: functions/Remove-DbaDatabase.ps1 [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	BEGIN
+	{
+		# Prompt ppl
+		$confirmpreference = "Low"
+		$databases = $psboundparameters.Databases
+		
+		if (-not $databases)
+		{
+			Write-Warning "You must select one or more databases to drop"
+			continue
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: Evil must be vanquished.
No using Continue!
Use Stop-Function here, add
`if (Test-FunctionInterrupt) { return }`
In subsequent blocks

File: functions/Remove-DbaDatabase.ps1 [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+		[parameter(Mandatory = $false)]
+		[object]$SqlCredential
+	)
+	
+	DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SqlCredential } }
+	
+	BEGIN
+	{
+		# Prompt ppl
+		$confirmpreference = "Low"
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: Bad form, instead set ConfirmImpact to a different level in Cmdletbinding

File: functions/Remove-DbaDatabase.ps1 [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+		}
+	}
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlInstance)
+		{
+			try
+			{
+				Write-Verbose "Connecting to $instance"
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: Needs some Write-Message magic

File: functions/Remove-DbaDatabase.ps1 [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+				$server = Connect-SqlServer -SqlServer $instance -SqlCredential $sqlcredential
+			}
+			catch
+			{
+				Write-Warning "Failed to connect to: $instance"
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: Classic case for Stop-Function

File: functions/Remove-DbaDatabase.ps1 [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+									InstanceName = $server.ServiceName
+									SqlInstance = $server.DomainInstanceName
+									Database = $db.name
+									Status = "Dropped"
+								}
+							}
+						}
+						catch
+						{
+							Write-Warning "Could not drop database $db on $server"
```
Comment (FriedrichWeinmann) [inline functions/Remove-DbaDatabase.ps1]: Needs some Write-Message magic (in warning mode)

File: functions/Remove-DbaDatabase.ps1 (hunk 3 above)
Comment (ctrlbold) [inline functions/Remove-DbaDatabase.ps1]: TRUE. im dumb. Thx

File: functions/Remove-DbaDatabase.ps1 [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+Does not prompt and swiftly removes containeddb on SQL Server sql2016
+#>
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[object[]]$SqlInstance,
+		[parameter(Mandatory = $false)]
+		[object]$SqlCredential
```
Comment (ctrlbold) [inline functions/Remove-DbaDatabase.ps1]: Done!

File: functions/Remove-DbaDatabase.ps1 (hunk 5 above)
Comment (ctrlbold) [inline functions/Remove-DbaDatabase.ps1]: Stop-Function continue or return?

**Pull request merged**